prboom2/src/p_enemy.c

```diff
@@ -3198,6 +3210,153 @@ void A_RadiusDamage(mobj_t *actor)
   P_RadiusAttack(actor, actor->target, actor->state->args[0], actor->state->args[1]);
 }
 
+//
+// A_NoiseAlert
+// Alerts nearby monsters (via sound) to the calling actor's target's presence.
+//
+void A_NoiseAlert(mobj_t *actor)
+{
+  if (!mbf21 || !actor->target)
+    return;
+
+  P_NoiseAlert(actor->target, actor);
+}
+
+//
+// A_HealChase
+// A parameterized version of A_VileChase.
+//   args[0]: State to jump to on the calling actor when resurrecting a corpse
+//   args[1]: Sound to play when resurrecting a corpse
+//
+void A_HealChase(mobj_t* actor)
+{
+  int state, sound;
+
+  if (!mbf21 || !actor)
+    return;
+
+  state = actor->state->args[0];
+  sound = actor->state->args[1];
+
+  if (!P_HealCorpse(actor, actor->info->radius, state, sound))
+    A_Chase(actor);
+}
+
+//
+// A_JumpIfHealthBelow
+// Jumps to a state if caller's health is below the specified threshold.
+//   args[0]: State to jump to
+//   args[1]: Health threshold
+//
+void A_JumpIfHealthBelow(mobj_t* actor)
+{
+  int state, health;
+
+  if (!mbf21 || !actor)
+    return;
+
+  state  = actor->state->args[0];
+  health = actor->state->args[1];
+
+  if (actor->health < health)
+    P_SetMobjState(actor, state);
+}
+
+//
+// A_JumpIfTargetInSight
+// Jumps to a state if caller's target is in line-of-sight.
+//   args[0]: State to jump to
+//
+void A_JumpIfTargetInSight(mobj_t* actor)
+{
+  int state;
+
+  if (!mbf21 || !actor)
+    return;
+
+  state = actor->state->args[0];
+
+  if (P_CheckSight(actor, actor->target))
+    P_SetMobjState(actor, state);
+}
+
+//
+// A_JumpIfTargetCloser
+// Jumps to a state if caller's target is closer than the specified distance.
+//   args[0]: State to jump to
+//   args[1]: Distance threshold
+//
+void A_JumpIfTargetCloser(mobj_t* actor)
+{
+  int state, distance;
+
+  if (!mbf21 || !actor)
+    return;
+
+  state    = actor->state->args[0];
+  distance = actor->state->args[1];
+
+  if (distance > P_AproxDistance(actor->x - actor->target->x,
+                                 actor->y - actor->target->y))
+    P_SetMobjState(actor, state);
+}
+
+//
+// A_JumpIfFlagsSet
+// Jumps to a state if caller has the specified thing flags set.
+//   args[0]: State to jump to
+//   args[1]: Flag(s) to check
+//
+void A_JumpIfFlagsSet(mobj_t* actor)
+{
+  int state;
+  unsigned int flags;
+
+  if (!mbf21 || !actor)
+    return;
+
+  state = actor->state->args[0];
+  flags = actor->state->args[1];
+
+  if (actor->flags & flags)
```

Comment:
Should this be `(actor->flags & flags) == flags`?

Reply:
Good catch -- is fixed now.